File: CS/EditorStatusViaCustomDraw/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;

namespace EditorStatusViaCustomDraw {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            dataTable1.Rows.Add("some value");
            dataTable1.Rows.Add(string.Empty);
            dataTable1.Rows.Add(DBNull.Value);
            dataTable1.Rows.Add("not empty");
        }

        private void gridView1_CustomDrawCell(object sender, DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs e) {
            if(e.Column.Name == "colCheckIndicator") {
                string text;
                bool isChecked;
                GridView grid = sender as GridView;
                if(grid.IsEditing && e.RowHandle == grid.FocusedRowHandle && grid.FocusedColumn.FieldName == "Column1") {
                    text = grid.ActiveEditor.Text;
                }
                else
                    text = grid.GetRowCellDisplayText(e.RowHandle, grid.Columns["Column1"]);
                isChecked = (text != null && text != string.Empty);
                DevExpress.XtraGrid.Views.Grid.ViewInfo.GridCellInfo cell = (DevExpress.XtraGrid.Views.Grid.ViewInfo.GridCellInfo)e.Cell;
                DevExpress.XtraEditors.ViewInfo.CheckEditViewInfo info = (DevExpress.XtraEditors.ViewInfo.CheckEditViewInfo)cell.ViewInfo;
                info.EditValue = isChecked;
            }
        }

        private void repositoryItemTextEdit1_EditValueChanged(object sender, EventArgs e) {
            GridView grid = gridView1;
            grid.InvalidateRowCell(grid.FocusedRowHandle, grid.Columns.ColumnByName("colCheckIndicator"));
        }
    }
}

File: VB/EditorStatusViaCustomDraw/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid

Namespace EditorStatusViaCustomDraw
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			dataTable1.Rows.Add("some value")
			dataTable1.Rows.Add(String.Empty)
			dataTable1.Rows.Add(DBNull.Value)
			dataTable1.Rows.Add("not empty")
		End Sub

		Private Sub gridView1_CustomDrawCell(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.RowCellCustomDrawEventArgs) Handles gridView1.CustomDrawCell
			If e.Column.Name = "colCheckIndicator" Then
				Dim text As String
				Dim isChecked As Boolean
				Dim grid As GridView = TryCast(sender, GridView)
				If grid.IsEditing AndAlso e.RowHandle = grid.FocusedRowHandle AndAlso grid.FocusedColumn.FieldName = "Column1" Then
					text = grid.ActiveEditor.Text
				Else
					text = grid.GetRowCellDisplayText(e.RowHandle, grid.Columns("Column1"))
				End If
				isChecked = (text IsNot Nothing AndAlso text <> String.Empty)
				Dim cell As DevExpress.XtraGrid.Views.Grid.ViewInfo.GridCellInfo = CType(e.Cell, DevExpress.XtraGrid.Views.Grid.ViewInfo.GridCellInfo)
				Dim info As DevExpress.XtraEditors.ViewInfo.CheckEditViewInfo = CType(cell.ViewInfo, DevExpress.XtraEditors.ViewInfo.CheckEditViewInfo)
				info.EditValue = isChecked
			End If
		End Sub

		Private Sub repositoryItemTextEdit1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles repositoryItemTextEdit1.EditValueChanged
			Dim grid As GridView = gridView1
			grid.InvalidateRowCell(grid.FocusedRowHandle, grid.Columns.ColumnByName("colCheckIndicator"))
		End Sub
	End Class
End Namespace